driver-benchmarks/src/main/com/mongodb/benchmark/framework/EvergreenBenchmarkResultWriter.java

```diff
@@ -22,53 +22,61 @@
 import java.io.FileWriter;
 import java.io.IOException;
-import java.io.Writer;
+import java.io.StringWriter;
 
 public class EvergreenBenchmarkResultWriter implements BenchmarkResultWriter {
 
     private static final String OUTPUT_FILE_SYSTEM_PROPERTY = "org.mongodb.benchmarks.output";
 
-    private final Writer writer;
+    private final StringWriter writer;
     private final JsonWriter jsonWriter;
 
     public EvergreenBenchmarkResultWriter()  {
-        try {
-            writer = new FileWriter(System.getProperty(OUTPUT_FILE_SYSTEM_PROPERTY));
-            jsonWriter = new JsonWriter(writer,
-                    JsonWriterSettings.builder().outputMode(JsonMode.RELAXED).indent(true).build());;
-            jsonWriter.writeStartDocument();
-            jsonWriter.writeStartArray("results");
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+        writer = new StringWriter();
+        jsonWriter = new JsonWriter(writer, JsonWriterSettings.builder().outputMode(JsonMode.RELAXED).indent(true).build());
+        jsonWriter.writeStartDocument();
+        jsonWriter.writeStartArray("results");
     }
 
     @Override
     public void write(final BenchmarkResult benchmarkResult) {
         jsonWriter.writeStartDocument();
-        jsonWriter.writeString("name", benchmarkResult.getName());
-        jsonWriter.writeStartDocument("results");
-        jsonWriter.writeStartDocument("1");
 
-        jsonWriter.writeDouble("ops_per_sec",
-                (benchmarkResult.getBytesPerIteration() / 1000000d) /
-                        (benchmarkResult.getElapsedTimeNanosAtPercentile(50) / 1000000000d));
+        jsonWriter.writeStartDocument("info");
+        jsonWriter.writeString("test_name", benchmarkResult.getName());
 
+        jsonWriter.writeStartDocument("args");
+        jsonWriter.writeInt32("threads", 1);
         jsonWriter.writeEndDocument();
         jsonWriter.writeEndDocument();
+
+        jsonWriter.writeStartArray("metrics");
+
+        jsonWriter.writeStartDocument();
+        jsonWriter.writeString("name","ops_per_sec" );
+        jsonWriter.writeDouble("value",
+                (benchmarkResult.getBytesPerIteration() / 1000000d) /
+                        (benchmarkResult.getElapsedTimeNanosAtPercentile(50) / 1000000000d));
+        jsonWriter.writeEndDocument();
+
+        jsonWriter.writeEndArray();
         jsonWriter.writeEndDocument();
     }
 
     @Override
     public void close() throws IOException {
         jsonWriter.writeEndArray();
         jsonWriter.writeEndDocument();
-        jsonWriter.close();
-        writer.close();
-        System.out.println("DONE");
+
+        try (FileWriter fileWriter = new FileWriter(System.getProperty(OUTPUT_FILE_SYSTEM_PROPERTY))) {
+            fileWriter.write(getJsonResultsArrayFromJsonDocument());
+        }
     }
 
-    public String getResults() {
-        return writer.toString();
+    private String getJsonResultsArrayFromJsonDocument() {
```

Comment:
This is the only awkward part.  JsonWriter does not allow writing a top-level array, so we have to write the array as the value of a field in a document.  This piece of code just strips off the outer document and field name and returns the JSON array.